File: runtime/Jlib/bint.java
package bigloo;

public class bint extends numeral
{
  public final int value;
  public static final bint BZERO= new bint( 0 );

  public bint( final int  value )
  {
    this.value = value;
  }

  public void write( final output_port  p )
  {
    p.write( Integer.toString( value ) );
  }
}

File: runtime/CSlib/bint.cs
using System;

namespace bigloo 
{
  public sealed class bint: numeral 
  {
    public readonly int value;

    public static readonly bint BZERO= new bint( 0 );

    public bint( int  value ) 
    {
      this.value= value;
    }

    public override void write( output_port  p ) 
    {
      p.write( value.ToString() );
    }
  }
}
